src/Filter/DateTime.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace TraderInteractive\Filter;
+
+use DateTimeInterface;
+use DateTimeZone as DateTimeZoneStandard;
+use DateTime as DateTimeStandard;
+use TraderInteractive\Exceptions\FilterException;
+
+/**
+ * A collection of filters for filtering strings into \DateTime objects.
+ */
+class DateTime
+{
+    /**
+     * Filters the given value into a \DateTime object.
+     *
+     * @param mixed                $value     The value to be filtered.
+     * @param boolean $allowNull              True to allow nulls through, and false (default) if nulls should
+     *                                        not be allowed.
+     * @param DateTimeZoneStandard $timezone  A \DateTimeZone object representing the timezone of $value.
+     *                                        If $timezone is omitted, the current timezone will be used.
+     *
+     * @return DateTimeStandard|null
+     *
+     * @throws FilterException if the value did not pass validation.
+     */
+    public static function filter($value, bool $allowNull = false, DateTimeZoneStandard $timezone = null)
+    {
+        if (self::valueIsNullAndValid($allowNull, $value)) {
+            return null;
+        }
+
+        if ($value instanceof \DateTime) {
+            return $value;
+        }
+
+        if (is_int($value) || ctype_digit($value)) {
+            $value = "@{$value}";
+        }
+
+        if (!is_string($value) || trim($value) == '') {
+            throw new FilterException('$value is not a non-empty string');
+        }
+
+        return new \DateTime($value, $timezone);
```

Comment:
you should be consistant with the `\DateTime` class. You are aliasing it as `DateTimeStandard` but using `\DateTime` here.